### PR TITLE
Use updated VisitorContext and make it optional

### PR DESCRIPTION
**Pull request by igorkravchenko**
Add changes regarding updated VisitorContext and provide ability to enter Visitor Context Asset ID via Settings.
Remove Mocks from GliaWidgetTests in favor of .Mock files.
Deprecate configuration related methods to take optional VisitorContext into account.

Note this PR is based on VisitorContext yet-to-be-released API. That is why it has WIP status.

MOB-1322

### CODE REVIEW

File: GliaWidgets/Configuration.Deprecated.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /// Deprecated. Initializes the configuration.
+    ///
+    /// - Parameters:
+    ///   - authorizationMethod: The site authorization method.
+    ///   - environment: The environment to use.
+    ///   - site: The site to use.
+    ///
+    @available(*, deprecated, message: "Deprecated. Please use Configuration(authorizationMethod:environment:site:visitorContext:) instead")
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply GliaWidgets/Configuration.Deprecated.swift]: Updated 578dc0346057e395e912779713ca4b7e4962fd15.

File: GliaWidgets/Configuration.Deprecated.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        )
+    }
+
+    /// Deprecated. Initializes the configuration.
+    ///
+    /// - Parameters:
+    ///   - authorizationMethod: The site authorization method.
+    ///   - environment: The environment to use.
+    ///   - site: The site to use.
+    ///
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply GliaWidgets/Configuration.Deprecated.swift]: Updated 578dc0346057e395e912779713ca4b7e4962fd15.

File: GliaWidgets/Configuration.Deprecated.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        )
+    }
+
+    /// Deprecated. Initializes the configuration.
+    ///
+    /// - Parameters:
+    ///   - appToken: The application token.
+    ///   - environment: The environment to use.
+    ///   - site: The site to use.
+    ///
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply GliaWidgets/Configuration.Deprecated.swift]: Updated 578dc0346057e395e912779713ca4b7e4962fd15.

File: GliaWidgets/Configuration.Deprecated.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+extension Configuration {
+    /// Deprecated. Initializes the configuration.
+    ///
+    /// - Parameters:
+    ///   - appToken: The application token.
+    ///   - apiToken: The API token.
+    ///   - environment: The environment to use.
+    ///   - site: The site to use.
+    ///
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply GliaWidgets/Configuration.Deprecated.swift]: Updated 578dc0346057e395e912779713ca4b7e4962fd15.

File: GliaWidgets/Configuration.Deprecated.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    @available(*, deprecated, message: "Deprecated. Please use Configuration(authorizationMethod:environment:site:visitorContext:) instead")
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply GliaWidgets/Configuration.Deprecated.swift]: Updated 578dc0346057e395e912779713ca4b7e4962fd15.

**Review by dukhovnyi — APPROVED** [review]

**Review by kuldar-daniel — APPROVED** [review]

**Pull request merged**